rmltraintfbbox/rmltraintfbbox/core.py

```diff
@@ -200,21 +260,100 @@ def train(ctx: click.Context, train: TrainInput):
                 experiment.log_asset(os.path.join(output_path, 'stats.json'))
                 for img in glob.glob(os.path.join(output_path, '*_curve_*.png')):
                     experiment.log_image(img)
-    except Exception:
-        metadata['validation_error'] = traceback.format_exc()
 
+    #export detection_model as SavedModel
+    saved_model_dir = os.path.join(model_dir,'export')
+    configproto = config_util.create_pipeline_proto_from_configs(configs)
+    export_inference_graph('image_tensor', configproto, model_dir, saved_model_dir)
+    
+    #zip files in export directory and add to extra_files
+    #shutil.make_archive(os.path.join(saved_model_dir,'export'), 'zip', model_dir, saved_model_dir)
+    shutil.make_archive(os.path.join(saved_model_dir,'export'), 'zip', model_dir, 'export')
+    extra_files.append(os.path.join(saved_model_dir, 'export.zip'))
+    saved_model_path = os.path.join(saved_model_dir, 'saved_model', 'saved_model.pb' )
+    
     if comet:
         experiment.log_asset_data(train.metadata, file_name="metadata.json")
 
     # export metadata locally
     with open(base_dir / 'metadata.json', 'w') as f:
         json.dump(train.metadata, f, indent=2)
-        
-    result = TrainOutput(Path(model_path), extra_files)
+    
+    extra_files = [Path(f) for f in extra_files]
+    result = TrainOutput(Path(saved_model_path), extra_files)
     return result
     
 
 ### HELPERS ###
+
+def load_fine_tune_checkpoint(
+    model, checkpoint_path, checkpoint_type, checkpoint_version, input_dataset,
+    unpad_groundtruth_tensors):
+
+#NOTE: Moved this function from OD API to add 'expect_partial' to the restore
```

Comment:
In the version of this function found in the OD API, the restore tries to "assert_existing_objects_matched" `ckpt.restore(checkpoint_path).assert_existing_objects_matched()`. For some reason, this assertion seems to throw an error. I had to modify the function to remove this assertion and changed the line to `ckpt.restore(checkpoint_path).expect_partial()`, which throws no error. From searching online it seemed like others had run into this problem too - this was my solution.

Reply:
Also, forgot to add that this assertion only fails for Centernet. I believe that this is because there is still some code related to Centernet missing from the OD API that would allow Centernet models to be restored from detection checkpoints, and not from classification checkpoints. So it is likely that in the future we can remove this and use the version of the function found in OD API.